src/qibolab/designs/basic.py

```diff
@@ -0,0 +1,47 @@
+from dataclasses import dataclass, field
+
+from qibolab.instruments.abstract import AbstractInstrument
+
+
+@dataclass
```

Comment:
Hi @stavros11,
I noticed you introduced the use of `dataclasses`.
@scarrazza, what is your view on it?
I think we should define criteria to guide us all on whether to use it or not, or in which cases. 
If we decide to use it, I believe we should review all project classes and apply consistently.

Reply:
Hi @aorgazf: I believe `dataclasses` might be helpful, but there is no need for an immediate thorough review.
You can start using them somewhere, decide if they are a good fit for your task, and apply them incrementally.

In the end, `dataclasses` are the standard library reduced version of [`attrs`](https://www.attrs.org/en/stable/), and one of the main motivations of `attrs` is to provide a shorter way to define classes, with less boilerplate and thus less maintenance burden.
But if you start changing everything at once, that is a non-negligible burden on its own :)

Reply:
> attributes/fields/properties

Though similar, they are three different things:
- *attributes* is what you have in mind, the data contained by the class (the [`getattr`](https://docs.python.org/3/library/functions.html?highlight=getattr#getattr) built-in, and [`__getattr__`](https://docs.python.org/3/reference/datamodel.html?highlight=getattribute#object.__getattr__) and [`__getattribute__`](https://docs.python.org/3/reference/datamodel.html?highlight=getattribute#object.__getattribute__) dunders, make it explicit)
- [`property`](https://docs.python.org/3/library/functions.html#property) is a built-in function, usually used as a decorator, to make a method being called with the same syntax of attribute look-up, so the pseudo-attributes generated in this way is what you might want to call *properties*
- *fields* are a concept strictly related to `dataclasses`, and essentially are the class attributes managed by the `dataclass` itself (but they might not coincide with the full set of instance attributes, since you are always free to add attributes at runtime to instances of unfrozen classes)

I can understand the confusion, however one gets familiar by using them.

> I think it would still be useful to have some guidance, to have some consistency, otherwise our code becomes a mess.

We can consider writing down a style guide, but this is an extra document to maintain and read, and might slow down refactoring.
Maybe it would be better just to discuss more with each other (we can periodically schedule some more informal meetings), to keep updated on our conventions and what we are doing. There are not an infinite amount of people working on the project ;)

Reply:
> Though similar, they are three different things:
> 
>     * _attributes_ is what you have in mind, the data contained by the class (the [`getattr`](https://docs.python.org/3/library/functions.html?highlight=getattr#getattr) built-in, and [`__getattr__`](https://docs.python.org/3/reference/datamodel.html?highlight=getattribute#object.__getattr__) and [`__getattribute__`](https://docs.python.org/3/reference/datamodel.html?highlight=getattribute#object.__getattribute__) dunders, make it explicit)
> 
>     * [`property`](https://docs.python.org/3/library/functions.html#property) is a built-in function, usually used as a decorator, to make a method being called with the same syntax of attribute look-up, so the pseudo-attributes generated in this way is what you might want to call _properties_
> 
>     * _fields_ are a concept strictly related to `dataclasses`, and essentially are the class attributes managed by the `dataclass` itself (but they might not coincide with the full set of instance attributes, since you are always free to add attributes at runtime to instances of unfrozen classes)
Very clarifying, thanks.

> We can consider writing down a style guide, but this is an extra document to maintain and read, and might slow down refactoring.
I don't think it would slow down our progress, on the contrary, it is likely going to help making the code right first time round.
The maintenance of the document is not hard, one can make small additions or corrections as any other file in the project when needed.
> There are not an infinite amount of people working on the project ;)
The project keeps growing and there are other teams in other locations with which we do not talk regularly.

Reply:
We don't have manpower at the moment to maintain a code style document which cannot be enforced programmatically. If new features are added to the code, more active reviews should be performed with common sense.